homeassistant/components/sharkiq/__init__.py

```diff
@@ -0,0 +1,119 @@
+"""Shark IQ Integration."""
+
+import asyncio
+
+import async_timeout
+from sharkiqpy import (
+    AylaApi,
+    SharkIqAuthError,
+    SharkIqAuthExpiringError,
+    SharkIqNotAuthedError,
+    get_ayla_api,
+)
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import API_TIMEOUT, COMPONENTS, DOMAIN, LOGGER
+from .update_coordinator import SharkIqUpdateCoordinator
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+
+class CannotConnect(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def async_setup(hass, config):
+    """Set up the sharkiq environment."""
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+
+
+async def async_connect_or_timeout(ayla_api: AylaApi) -> bool:
+    """Connect to vacuum."""
+    try:
+        with async_timeout.timeout(API_TIMEOUT):
+            LOGGER.debug("Initialize connection to Ayla networks API")
+            await ayla_api.async_sign_in()
+    except SharkIqAuthError as exc:
+        LOGGER.error("Authentication error connecting to Shark IQ api", exc_info=exc)
```

Comment:
we shouldnt print stack trace since its an expected error
```suggestion
        LOGGER.error("Authentication error connecting to Shark IQ api")
```